### DISCUSSION TITLE
Server Side Rendering and Scaling in NextJS

### DISCUSSION THREAD

**Discussion by mshahins** [vercel/next.js#44267] [Help]
We are in process of setting up new architecture for our new application and using react. I have read about React and Next.Js. Advantage that Next.js brings is SSR (Server side Rendering). As I understand as part of SSR rendering of Java Script and generation of HTML will happen on server side and HTML will be served to UI which would be displayed by UI. 

My questions are below. 

1. I have very dynamic web site; i.e. there are lot of grids and data intensive application which changes based on user search criteria. Is NextJs good option for this ?
2. If NextJS is good option for Item 1 then do we have to think about Scaling Hardware ? The reason I am asking is if we have hundreds of users using this web site where NextJS is enabled then lot of load will increase on server as most of the heavy lifting is happening on server side.

I would really appreciate your feedback on this and experience on this. And any other scenarios to consider while developing very dynamic webstie which is very data intensive.

Regards
Mike

**Comment by GabenGar** [comment]
Very dynamic websites are created through client render. SSR is only needed for SEO, so there is literally no point in using it for your usecase (user-specific pages). Not sure what kind of architecture or scalability you are talking about, when you don't even see the problem of rendering uncacheable html on server.
